oneandone/src/test/java/org/apache/jclouds/oneandone/rest/features/MonitoringCenterApiMockTest.java

```diff
@@ -0,0 +1,157 @@
+
+   private MonitoringCenterApi monitoringCenterApi() {
+      return api.monitoringCenterApi();
+   }
+
+   @Test
+   public void testList() throws InterruptedException {
+      server.enqueue(
+              new MockResponse().setBody(stringFromResource("/monitoringcenters/list.json"))
+      );
+
+      List<MonitoringCenter> result = monitoringCenterApi().list();
+
+      assertNotNull(result);
+      assertEquals(result.size(), 2);
+
+      assertEquals(server.getRequestCount(), 1);
+      assertSent(server, "GET", "/monitoring_center");
+   }
+
+   @Test
+   public void testList404() throws InterruptedException {
+      server.enqueue(
+              new MockResponse().setResponseCode(404));
+
+      List<MonitoringCenter> result = monitoringCenterApi().list();
+
+      assertEquals(result.size(), 0);
+
+      assertEquals(server.getRequestCount(), 1);
+      assertSent(server, "GET", "/monitoring_center");
+   }
+
+   @Test
+   public void testListWithOption() throws InterruptedException {
+      server.enqueue(
+              new MockResponse().setBody(stringFromResource("/monitoringcenters/list.options.json"))
+      );
+      GenericQueryOptions options = new GenericQueryOptions();
+      options.options(0, 0, null, "New", null);
+      List<MonitoringCenter> result = monitoringCenterApi().list(options);
+
+      assertEquals(result.size(), 2);
+      assertEquals(server.getRequestCount(), 1);
+      assertSent(server, "GET", "/monitoring_center?q=New");
+   }
+
+   @Test
+   public void testListWithOption404() throws InterruptedException {
+      server.enqueue(
+              new MockResponse().setResponseCode(404));
+
+      GenericQueryOptions options = new GenericQueryOptions();
+      options.options(0, 0, null, "New", null);
+      List<MonitoringCenter> result = monitoringCenterApi().list(options);
+
+      assertEquals(result.size(), 0);
+      assertEquals(server.getRequestCount(), 1);
+      assertSent(server, "GET", "/monitoring_center?q=New");
+   }
+
+   public void testGetCustomPeriod() throws InterruptedException, ParseException {
+      server.enqueue(
+              new MockResponse().setBody(stringFromResource("/monitoringcenters/get.json"))
+      );
+      GenericDateQueryOptions options = new GenericDateQueryOptions();
+      String startStr = "11-11-2012";
+      String endStr = "11-11-2013";
+      DateFormat dateFormat = new SimpleDateFormat("dd-MM-yyyy");
+      Date end = dateFormat.parse(endStr);
+      Date start = dateFormat.parse(startStr);
+      options.CustomPeriod(start, end);
+      MonitoringCenter result = monitoringCenterApi().get("serverId", options);
+
+      assertNotNull(result);
+      assertEquals(server.getRequestCount(), 1);
+      assertSent(server, "GET", "/monitoring_center/serverId?period=CUSTOM&start_date=" + URLEncoder.encode(FormatDateToISO8601(start)) + "&end_date=" + URLEncoder.encode(FormatDateToISO8601(end)));
```

Comment:
Put the expected date as a string literal, to make sure we are serializing the date as expected.